### PR TITLE
add tests for v1/config API

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        def flow = flowHelper.randomFlow(switchPair)
+        flow.encapsulationType = null
+        northbound.addFlow(flow)
+
+        then: "Flow is created with correct encapsulation type"
+        northbound.getFlow(flow.id).encapsulationType == currentConfiguration.flowEncapsulationType
+
+        cleanup: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+    }
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy]: The spec is obviously missing a test case that would update the default encap type and check that flow creates with an updated new value. The above test may be just a coincidence. How do you feel about merging 1st and 2nd tests into one and checking config update together with creating a flow?

**Comment by rtretyak** [comment]
Also didn't quite get why it's marked as low priority, can you share you thoughts?

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import org.openkilda.messaging.model.system.KildaConfigurationDto
+import org.openkilda.model.FlowEncapsulationType
+
+import org.springframework.http.HttpStatus
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+import spock.lang.Shared
+
+@Tags(LOW_PRIORITY)
+@Narrative("""Kilda configuration is a special lever that allows to default flow encapsulation type while creating.
```
[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy]: thx, fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        given: "Current default flow encapsulation type"
+        def currentConfiguration = northbound.getKildaConfiguration()
+
+        when: "Create a flow without encapsulation type"
+        def switchPair = topologyHelper.getRandomSwitchPair()
+        def flow = flowHelper.randomFlow(switchPair)
+        flow.encapsulationType = null
+        northbound.addFlow(flow)
+
+        then: "Flow is created with correct encapsulation type"
```
[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy]: agree, fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy]: I had second thoughts about missing test case while writing this spec. 
but now I see it have to be added. 
So: 
- missing test case was added with `HARDWARE` tag.
- 1st and 2nd tests were merged.
- this test covers v2 only

Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy]: I don't like this test. How do you know that system is not _always_ creating a vxlan flow when encap type is missing? As for me the steps should be as follows
```
1. Create a flow without encap
2. Check that flow has transit_vlan (current default)
3. update current default encap type to vxlan
4. create one more flow
5. Check that the encap type has also changed to vxlan
```

Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/configuration/ConfigurationSpec.groovy]: I can't argue with that (I tried to save time for running this test)

Fixed.

**Comment by andriidovhan** [comment]
> Also didn't quite get why it's marked as low priority, can you share you thoughts?

We don't have a lot of changes in this component.
I guess it is acceptable to run this test during RC testing only

**Pull request merged**